Shared/Samples/Set reference scale/SetReferenceScaleView.swift

```diff
@@ -0,0 +1,166 @@
+    @State private var settingsPopoverIsPresented = false
+    
+    var body: some View {
+        MapViewReader { mapViewProxy in
+            MapView(map: map)
+                .onScaleChanged { mapScale = $0 }
+                .toolbar {
+                    ToolbarItem(placement: .bottomBar) {
+                        Button("Map Settings") {
+                            settingsPopoverIsPresented = true
+                        }
+                        .popover(isPresented: $settingsPopoverIsPresented) { [mapScale] in
+                            MapSettingsView(map: map, mapScale: mapScale ?? 0) { scale in
+                                // Sets the map's scale to the selected reference scale
+                                // when the "Set to Reference Scale" button is tapped.
+                                Task {
+                                    await mapViewProxy.setViewpointScale(scale)
+                                }
+                            }
+                            .presentationDetents([.fraction(0.5)])
+                            .frame(idealWidth: 320, idealHeight: 380)
+                        }
+                    }
+                }
+        }
+    }
+}
+
+private extension SetReferenceScaleView {
+    /// The view containing settings controls for a map.
+    struct MapSettingsView: View {
+        /// The map.
+        let map: Map
+        
+        /// A binding to the scale of the map.
+        let mapScale: Double
+        
+        /// The action to set the map scale, i.e, when the "Set to Reference Scale" button is pressed.
+        let setMapScaleAction: (_ scale: Double) -> Void
+        
+        /// The action to dismiss the view.
+        @Environment(\.dismiss) private var dismiss
+        
+        /// The reference scale option selected in the picker.
+        @State private var selectedReferenceScale: Double = 250_000
+        
+        /// The options for the reference scale picker
+        private let referenceScaleOptions: [Double] = [500_000, 250_000, 100_000, 50_000]
+        
+        var body: some View {
+            NavigationStack {
+                Form {
+                    Section {
+                        Picker("Reference Scale", selection: $selectedReferenceScale) {
+                            ForEach(referenceScaleOptions, id: \.self) { option in
+                                Text("1:\(option, format: .number.rounded(increment: 1))")
+                            }
+                        }
+                        .onChange(of: selectedReferenceScale) { map.referenceScale = $0 }
+                        
+                        NavigationLink("Layers") {
+                            Form {
+                                ForEach(map.operationalLayers as! [FeatureLayer], id: \.id) { layer in
+                                    ScalesSymbolsToggle(layer: layer)
+                                }
+                            }
```

Comment:
```suggestion
                            List(map.operationalLayers as! [FeatureLayer], id: \.id) { layer in
                                ScalesSymbolsToggle(layer: layer)
                            }
```

Reply:
Still a learning when to use List/Form. Wouldn't a form be used here since the content contains settings?

Reply:
Yes, which is why all of the various setting views are contained in a form. From [the doc](https://developer.apple.com/documentation/swiftui/form) (emphasis mine):

> SwiftUI applies platform-appropriate styling to views contained inside a form, to group them together. **Form-specific styling applies to things like** buttons, toggles, labels, **lists**, and more.

So by being enclosed in a form (being a subview of a form), the list gets form styling automatically. No need to put a form in a form just to get form styling.

Does that make sense?

Reply:
Yes, thank you!